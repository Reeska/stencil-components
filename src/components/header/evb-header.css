:host {
  align-items: center;
  display: flex;
}
.header {
  line-height: normal;
  margin: 0;
  padding: 1rem 0;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

h1 {
  font-style: normal;
  font-size: 3em;
  font-weight: 600;
}
h2 {
  font-size: 1.5em;
  font-weight: 500;
  line-height: 1;
}
h3 {
  font-weight: 400;
}
