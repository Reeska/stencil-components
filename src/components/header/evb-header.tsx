import { Component, ComponentInterface, Prop } from '@stencil/core';

const MIN_HEADING = 1;
const MAX_HEADING = 6;

/**
 * Container for buttons
 */
@Component({
  tag: 'evb-header',
  styleUrl: 'evb-header.css',
  shadow: true
})
export class EvbButtonBar implements ComponentInterface {

  @Prop() heading!: number | string;

  constructor() {
    this.validateHeading(this.heading);
  }

  render() {
    // tslint:disable-next-line: no-any variable-name
    const TagType = `h${this.heading}` as any;
    return (
      <TagType class='header'>
        <slot />
      </TagType>
    );
  }

  /**
   * Validates that the heading is a valid heading (between 1 and 6)
   *
   * @throws when heading is not valid
   */
  private validateHeading(heading: number | string) {
    const number: number = typeof heading === 'number' ? heading : parseInt(heading, 10);
    if (number > MAX_HEADING || number < MIN_HEADING) {
      throw new Error('Heading must be between 1 and 6');
    }
  }
}
