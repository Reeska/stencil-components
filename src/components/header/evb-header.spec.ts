import { EvbHeader } from './evb-header';

describe('EvbRange', () => {
  let component: EvbHeader;

  beforeEach(() => component = new EvbHeader());

  it('should render', () => {
    expect(component.render()).toMatchSnapshot();
  });
});
