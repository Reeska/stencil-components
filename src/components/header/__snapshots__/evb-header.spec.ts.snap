// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvbRange should render 1`] = `
Object {
  "$attrs$": Object {
    "class": "header",
  },
  "$children$": Array [
    Object {
      "$attrs$": null,
      "$children$": null,
      "$elm$": null,
      "$flags$": 0,
      "$key$": null,
      "$name$": null,
      "$tag$": "slot",
      "$text$": null,
    },
  ],
  "$elm$": null,
  "$flags$": 0,
  "$key$": null,
  "$name$": null,
  "$tag$": "hundefined",
  "$text$": null,
}
`;
